machine/v1/types_nutanixprovider.go

```diff
@@ -0,0 +1,117 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NutanixMachineProviderConfig is the Schema for the nutanixmachineproviderconfigs API
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type NutanixMachineProviderConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// cluster is to identify the cluster (the Prism Element under management
+	// of the Prism Central), in which the Machine's VM will be created.
+	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	// +kubebuilder:validation:Required
+	Cluster NutanixResourceIdentifier `json:"cluster"`
+
+	// image is to identify the rhcos image uploaded to the Prism Central (PC)
+	// The image identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	// +kubebuilder:validation:Required
+	Image NutanixResourceIdentifier `json:"image"`
+
+	// subnet is to identify the cluster's network subnet to use for the Machine's VM
+	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
+	// or using the prism_central API.
+	// +kubebuilder:validation:Required
+	Subnet NutanixResourceIdentifier `json:"subnet"`
+
+	// vcpusPerSocket is the number of vCPUs per socket of the VM
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
+	VcpusPerSocket int32 `json:"vcpusPerSocket"`
+
+	// vcpuSockets is the number of vCPU sockets of the VM
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
+	VcpuSockets int32 `json:"vcpuSockets"`
```

Comment:
CPU should be capitalised on both of these IMO
```suggestion
	// vcpusPerSocket is the number of vCPUs per socket of the VM
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:Minimum=1
	VCPUsPerSocket int32 `json:"vcpusPerSocket"`

	// vcpuSockets is the number of vCPU sockets of the VM
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:Minimum=1
	VCPUSockets int32 `json:"vcpuSockets"`
```